app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/controller/CafeteriaMenuManager.java

```diff
@@ -86,72 +86,48 @@ private void onDownloadSuccess(@NonNull CafeteriaResponse response) {
-     * if he actually goes to that specific mensa. The alarm is then stored and scheduled, if it's not
-     * scheduled already.
-     *
-     * @param completeReschedule True if all currently scheduled alarms should be discarded, False if not
-     */
-    public void scheduleFoodAlarms(boolean completeReschedule) {
-        FavoriteFoodAlarmStorage favoriteFoodAlarmStorage = FavoriteFoodAlarmStorage.getInstance()
-                .initialize(mContext);
-        if (completeReschedule) {
-            favoriteFoodAlarmStorage.cancelOutstandingAlarms();
+        for (DateTime menuDate : menuDates) {
+            LocalTime weekdayNotificationTime = settings.retrieveLocalTime(menuDate);
+            if (weekdayNotificationTime != null) {
+                DateTime notificationTime = weekdayNotificationTime.toDateTimeToday();
+                notificationTimes.add(notificationTime);
+            }
         }
 
-        List<String> dates = favoriteDishDao.getFavouriteDishDates();
-
-        for (String date : dates) {
-            favoriteFoodAlarmStorage.scheduleAlarm(date);
-        }
+        NotificationScheduler scheduler = new NotificationScheduler(mContext);
+        scheduler.scheduleAlarms(NotificationType.CAFETERIA, notificationTimes);
     }
 
     /**
-     * This method returns all the mensas serving favorite dishes at a given day and their unique
-     * dishes
+     * Returns all the favorite dishes that a particular mensa serves on the specified date.
      *
-     * @param dayMonthYear String with ISO-Date (yyyy-mm-dd)
+     * @param mensaId The Cafeteria for which to return the favorite dishes served
+     * @param date The date for which to return the favorite dishes served
      * @return the favourite dishes at the given date
      */
-    public HashMap<Integer, HashSet<CafeteriaMenu>> getServedFavoritesAtDate(String dayMonthYear) {
-        HashMap<Integer, HashSet<CafeteriaMenu>> cafeteriaServedDish = new HashMap<>();
+    public List<CafeteriaMenu> getFavoriteDishesServed(int mensaId, DateTime date) {
+        List<CafeteriaMenu> results = new ArrayList<>();
+        String dateString = DateTimeUtils.INSTANCE.getDateString(date);
+
+        List<CafeteriaMenu> upcomingServings = favoriteDishDao.getFavouritedCafeteriaMenuOnDate(dateString);
 
-        List<CafeteriaMenu> upcomingServings = favoriteDishDao.getFavouritedCafeteriaMenuOnDate(dayMonthYear);
         for (CafeteriaMenu upcomingServing : upcomingServings) {
-            int mensaId = upcomingServing.getCafeteriaId();
-            HashSet<CafeteriaMenu> servedAtCafeteria;
-            if (cafeteriaServedDish.containsKey(mensaId)) {
-                servedAtCafeteria = cafeteriaServedDish.get(mensaId);
-            } else {
-                servedAtCafeteria = new HashSet<>();
-                cafeteriaServedDish.put(mensaId, servedAtCafeteria);
+            int cafeteriaId = upcomingServing.getCafeteriaId();
+
+            if (cafeteriaId == mensaId) {
```

Comment:
naming...
Could we consider something like `queriedMensaId` and `currentMensaId`?